### PR TITLE
New branch

### PR DESCRIPTION
**Pull request by XeniaSp**

### CODE REVIEW

File: .vagrant/machines/node-1/virtualbox/action_provision [hunk 1]
```diff
@@ -1 +0,0 @@
-1.5:f0e5c919-ccdc-4303-8bd7-0b4ed6799e7c
```
Comment (mspanakis) [inline .vagrant/machines/node-1/virtualbox/action_provision:1]: Exeis kanei submit olo to .vagrant directory. Prepei na to sviseis

File: .vagrant/machines/node-1/virtualbox/private_key [hunk 2]
```diff
@@ -1,27 +0,0 @@
------BEGIN RSA PRIVATE KEY-----
```
Comment (mspanakis) [inline .vagrant/machines/node-1/virtualbox/private_key:1]: Kai pali, olo to .vagrant. Episis, private kleidia pote pote pote submit.

File: Roles/Firewall/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+---
+# tasks file for Firewall
+
+  - name: Install ufw from apt
+    apt:  name=ufw  state=present
+  
+  - name: Generate the rules
+    template: src={{ item }}.j2
+              dest=/lib/ufw/{{ item }}
```
Comment (mspanakis) [inline Roles/Firewall/tasks/main.yml:9]: Edw, theleis tin kataliksi j2? Enas kaliteros tropos tha itan na kratas sto directory templates to idio structure. Px:

templates/lib/ufw/user.rules.j2

Opote edw sto src kai dest tha mporouses na eixes:
src: "{{ item }}"
dest: "/{{ item }}"

Epeisis, svise ta = kai kanta : se new grammi
px template: src={{item }}.j2 ginetai:

template:
   src: "{{ item }}"mpla mpla

File: Roles/Firewall/tests/test.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+- hosts: localhost
+  remote_user: root
+  roles:
+    - Firewall
```
Comment (mspanakis) [inline Roles/Firewall/tests/test.yml:5]: Edw kati paraponietai oti den exei new line. Ti testing kaneis me afto edw? 

To inventory apo panw, to xreiazesai? den xrisimopoieis sto .vagrant inventory?

File: Roles/Firewall/vars/main.yml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+---
+# vars file for Firewall
+http_port=80
```
Comment (mspanakis) [inline Roles/Firewall/vars/main.yml:3]: afta tha mporousan na mpoun sto defaults/main.yml . Se fasi, ta defaults einai afta alla an thes gia kapoio logo allo port, mporeis na to kaneis overwrite

File: Roles/Setup/README.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+Role Name
+=========
+
+A brief description of the role goes here.
```
Comment (mspanakis) [inline Roles/Setup/README.md:4]: Grapse kati re :P

File: Roles/Setup/tasks/main.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
+---
+# tasks file for Setup
+
+  - name: Install pip from apt
+    apt: name=python-pip state=present
+
+  - name: Update APT package cache
+    apt: update_cache=yes
```
Comment (mspanakis) [inline Roles/Setup/tasks/main.yml:8]: na akoloutheis to structure pou sou eipa parapanw....oxi name=mpla alla
name: mpla
state: present

File: Roles/Setup/tasks/main.yml [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+  - name: Run apt-get upgrade
+    apt: upgrade=full
+
+  - name: Username and Password
+      user: user
+      password: "{{password}}"
```
Comment (mspanakis) [inline Roles/Setup/tasks/main.yml:15]: ti prospatheis edw?

File: playbook.yml [hunk 9]
```diff
@@ -6,25 +6,7 @@
...
-
-  - hosts: node-3
-    become: True
-
-    host_vars: node-3
-
-  - hosts: node-4
-    become: True
-
-    host_vars: node-4
```
Comment (mspanakis) [inline playbook.yml:27]: wraia

File: .vagrant/machines/node-1/virtualbox/action_provision (hunk 1 above)
Comment (mspanakis) [reply .vagrant/machines/node-1/virtualbox/action_provision:1]: Mporeis na to prostheseis sto .gitignore directory wste na min emfanizetai san not tracked or modified

Comment (mspanakis) [reply .vagrant/machines/node-1/virtualbox/action_provision:1]: Epeisis, twra eida to onoma tou branch. Dine pio perigrafiko onoma giati to NewBranch den mou leei tpt.. kai xrisimopoiise mikra gia ta onomata twn fakelwn, poli pio efkolo typing :)

**Review by mspanakis — COMMENTED** [review]
Here are the comments

File: Roles/Firewall/tests/test.yml (hunk 4 above)
Comment (XeniaSp) [reply Roles/Firewall/tests/test.yml:5]: auto edw to kommati den thumamai na to ekana kan! den to exw shmeiwsei kan sto shmeiwmatario pou krataw thn methodologia pou akolouthw gia na mhn ta ksexasw.... ti mporei na einai? skeftomai na to svhsw

File: Roles/Setup/tasks/main.yml (hunk 8 above)
Comment (XeniaSp) [inline Roles/Setup/tasks/main.yml]: otan ginetai to tade username na einai me to tade password.. axrhsto les?

Comment (XeniaSp) [reply Roles/Setup/tasks/main.yml:15]: vasika koita na deis... exw valei sto arxeio me ta hosts : /Project/host_vars gia kathe node (kathe ea apo ta virtual machines mou)mesa ena hashed password. thelw me username : user kathe fora na zhtaei kwdiko kai na mporei na mpei o xrhsths mesa.. twra omws edw auto uparxei sto Project/Setup/tasks/main.yml kai oi kwdikoi gia kathe ena einai sto Project/host_vars/node_(tade). pws tha to kanoume?

Comment (XeniaSp) [reply Roles/Setup/tasks/main.yml:15]: thumasai pou ekana hashed password kai eftiaksa ena arxeio host_vars mesa sto opoio gia kathe node eixe to password. mesa ekei skeftomai na eisagw akoma mia metavlhth.dld na einai kapws etsi:
user=node-1
password=jcnsdohyfwjodsk
etsi gia kathe node. 
kai meta edw sto setup/tasks/main.yml na einai etsi:

-name: Username and Password
    user: {{ user }}
    password: {{ password }}

tha einai swsto?

Comment (mspanakis) [reply Roles/Setup/tasks/main.yml:15]: Opws to eixes prin, tha exeis ousiastika enan xristi "user" se kathe vm me diaforetiko password? Me ton idio tropo mporeis na valeis kai diaforetiko xristi per host. Thes diaforetikous xristes se kathe vm?

**Pull request closed without merging**